lib/core/include/scipp/core/element/arithmetic.h

```diff
@@ -193,7 +193,8 @@ constexpr auto mod = overloaded{
     [](const units::Unit &a, const units::Unit &b) { return a % b; }};
 
 constexpr auto mod_equals =
-    overloaded{arg_list<int64_t, int32_t, std::tuple<int64_t, int32_t>>,
+    overloaded{remainder_types_t{}, transform_flags::expect_no_variance_arg<0>,
+               transform_flags::expect_no_variance_arg<1>,
```

Comment:
```suggestion
    overloaded{arg_list<double, float, int64_t, int32_t, std::tuple<double, float>,
             std::tuple<float, double>, std::tuple<int64_t, int32_t>,
             std::tuple<int32_t, int64_t>>,
    transform_flags::expect_no_variance_arg<0>,
    transform_flags::expect_no_variance_arg<1>,
```
something like this? (allow mixing int32 with int64, float with double, but not float with int)

Reply:
Are you sure we should allow `<int32_t, int64_t>`? It would need to convert an `int64` to and `int32` in the output? Same for `<float, double>`.

Reply:
I think that's what numpy allows, but we don't have to do everything like numpy

Reply:
For now I think it makes sense to use what Numpy allows as a default if we don't have a particular reason do something different.

Reply:
Yes, I agree going with NumPy makes sense in this case.